images/vector/tests/main.tf

```diff
@@ -8,25 +8,36 @@ variable "digest" {
   description = "The image digest to run tests over."
 }
 
+variable "helm_chart_tag" {
+  description = "Whether to pull a specific version of the helm chart for testing."
+  default     = ""
+}
+
 data "oci_string" "ref" { input = var.digest }
 
+resource "random_id" "hex" { byte_length = 4 }
+
 resource "helm_release" "vector" {
-  name       = "vector"
-  repository = "https://helm.vector.dev"
-  chart      = "vector"
+  name             = "vector-${random_id.hex.hex}"
+  repository       = "https://helm.vector.dev"
+  chart            = "vector"
+  create_namespace = true
+  namespace        = "vector-${random_id.hex.hex}"
+  timeout          = 120
 
   values = [jsonencode({
     localpv = {
       image = {
         registry   = join("", [data.oci_string.ref.registry, "/"])
         repository = data.oci_string.ref.repo
-        tag        = data.oci_string.ref.pseudo_tag
+        tag        = var.helm_chart_tag != "" ? var.helm_chart_tag : data.oci_string.ref.pseudo_tag
```

Comment:
Hmm, I don't believe this will work.

Reply:
Consider: we don't publish a tag until tests (including this) have completed, so the only times this would actually qualify the image in question would be if the tag incidentally applied to the same image that we'd previously let through.

Reply:
Thanks @mattmoor, is your feedback in relation to this specifically (i.e before my amendments):

```bash
tag = data.oci_string.ref.pseudo_tag
```

Also bump @srishtih

Reply:
ACK thanks Matt for calling out. I think in other such examples we use the digest, so applied the same to here